pkg/netflow/flowaggregator/flowaccumulator.go

```diff
@@ -0,0 +1,100 @@
+package flowaggregator
+
+import (
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/netflow/common"
+)
+
+var timeNow = time.Now
+
+// floWrapper contains flow information and additional flush related data
+type flowWrapper struct {
+	flow                *common.Flow
+	nextFlush           time.Time
+	lastSuccessfulFlush time.Time
+}
+
+// flowAccumulator is used to accumulate aggregated flows
+type flowAccumulator struct {
+	flows             map[string]flowWrapper
+	mu                sync.Mutex
+	flowFlushInterval time.Duration
+	flowContextTTL    time.Duration
+}
+
+func newFlowWrapper(flow *common.Flow) flowWrapper {
+	now := timeNow()
+	return flowWrapper{
+		flow:      flow,
+		nextFlush: now,
+	}
+}
+
+func newFlowAccumulator(aggregatorFlushInterval time.Duration) *flowAccumulator {
+	return &flowAccumulator{
+		flows:             make(map[string]flowWrapper),
+		flowFlushInterval: aggregatorFlushInterval,
+		flowContextTTL:    aggregatorFlushInterval * 5,
+	}
+}
+
+// flush will flush specific flow context (distinct hash) if nextFlush is reached
+// once a flow context is flushed nextFlush will be updated to the next flush time
+// Specific flow context in `flowAccumulator.flows` map will be deleted if `flowContextTTL`
+// to avoid keeping flow context that are not seen anymore.
+func (f *flowAccumulator) flush() []*common.Flow {
+	f.mu.Lock()
+	defer f.mu.Unlock()
+
+	var flows []*common.Flow
+	for key, flow := range f.flows {
+		now := timeNow()
+		if flow.nextFlush.After(now) {
+			continue
+		}
+		if flow.flow != nil {
+			flows = append(flows, flow.flow)
+			flow.lastSuccessfulFlush = now
+			flow.flow = nil
+		} else if flow.lastSuccessfulFlush.Add(f.flowContextTTL).Before(now) {
+			// delete flow wrapper if there is no successful flushes since `flowContextTTL`
+			delete(f.flows, key)
+			continue
+		}
+		flow.nextFlush = flow.nextFlush.Add(f.flowFlushInterval)
+		f.flows[key] = flow
+	}
+	return flows
+}
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
func (f *flowAccumulator) flush() []*common.Flow {
	f.mu.Lock()
	defer f.mu.Unlock()

	var flowsToFlush []*common.Flow
	for key, flow := range f.flows {
		now := timeNow()
		if flow.nextFlush.After(now) {
			continue
		}
		if flow.flow != nil {
			flowsToFlush = append(flowsToFlush, flow.flow)
			flow.lastSuccessfulFlush = now
			flow.flow = nil
		} else if flow.lastSuccessfulFlush.Add(f.flowContextTTL).Before(now) {
			// delete flow wrapper if there is no successful flushes since `flowContextTTL`
			delete(f.flows, key)
			continue
		}
		flow.nextFlush = flow.nextFlush.Add(f.flowFlushInterval)
		f.flows[key] = flow
	}
	return flowsToFlush
}
```